test/yapi_network_test.rb

```diff
@@ -0,0 +1,374 @@
+#! /usr/bin/env rspec
+
+require_relative "test_helper"
+
+require "yast"
+
+Yast.import "YaPI::NETWORK"
+Yast.import "NetworkInterfaces"
+Yast.import "Routing"
+Yast.import "IP"
+Yast.import "Service"
+Yast.import "DNS"
+Yast.import "LanItems"
+
+describe Yast::YaPI::NETWORK do
+  include YaPINetworkStub
+
+  describe ".Write" do
+
+    context "with routes" do
+      # Mock the system to do not break anything
+      before do
+        allow(Yast::Routing).to receive(:Read)
+        allow(Yast::Routing).to receive(:Routes)
+        allow(Yast::Routing).to receive(:Write)
+      end
+
+      subject { Yast::YaPI::NETWORK.Write("route" => route) }
+
+      let(:success) { { "error" => "", "exit" => "0" } }
+
+      context "with no gateway in the default route" do
+        let(:route) { { "default" => nil } }
+
+        it "returns success" do
+          expect(subject).to eq success
+        end
+
+        it "empties the routes" do
+          expect(Yast::Routing).to receive(:Routes=).with []
+          subject
+        end
+      end
+
+      context "with empty gateway in the default  route" do
+        let(:route) { { "default" => { "via" => "" } } }
+
+        it "returns success" do
+          expect(subject).to eq success
+        end
+
+        it "empties the routes" do
+          expect(Yast::Routing).to receive(:Routes=).with []
+          subject
+        end
+      end
+
+      context "with a gateway in the default route" do
+        let(:route) { { "default" => { "via" => "TheIP" } } }
+        before do
+          expect(Yast::IP).to receive(:Check4).with("TheIP").and_return valid
```

Comment:
I didn't get this mocking. Why to mock such simple function as ```Check4``` for validating obviously invalid input? You can use any ip instead of the string ```TheIP```. Moreover there are predefined IPs in a RFC (if think it is RFC5737) which can be used in documentation / examples. Such IP's must not be used in real word.